lib/reversetunnel/srv.go

```diff
@@ -1052,22 +1052,22 @@ func newRemoteSite(srv *server, domainName string, sconn ssh.Conn) (*remoteSite,
 	return remoteSite, nil
 }
 
-// DELETE IN: 5.1.0.
+// DELETE IN: 7.0.0.
 //
-// isOldCluster checks if the cluster is older than 5.0.0.
+// isOldCluster checks if the cluster is older than 6.0.0.
 func isOldCluster(ctx context.Context, conn ssh.Conn) (bool, error) {
 	version, err := sendVersionRequest(ctx, conn)
 	if err != nil {
 		return false, trace.Wrap(err)
 	}
 
-	// Return true if the version is older than 5.0.0, the check is actually for
-	// 4.5.0, a non-existent version, to allow this check to work during development.
+	// Return true if the version is older than 6.0.0, the check is actually for
+	// 5.9.0, a non-existent version, to allow this check to work during development.
```

Comment:
Why wouldn't `6.0.0` work in the check?
We don't actually have to have it released for the semver comparison to work, right?

Reply:
@awly I think this is to handle the development scenario where you have two 6.0 clusters e.g. 6.0.0 and 6.0.0-dev.1 or something. If we check against 6.0.0 here, then it would be considered as "old". Unless I'm missing something.

Reply:
Would `6.0.0-dev.1` be considered old? Or `6.0.0`?
I still don't understand how.

Reply:
@awly So let's say we have 2 clusters - `6.0.0` (root) and `6.0.0-rc.1` (leaf). If we compare with `6.0.0` here then `6.0.0-rc.1` cluster will be considered "old" (since 6.0.0-rc.1 < 6.0.0) and the cache will be setup incorrectly.

Technically, once we release 6.0.0 GA this shouldn't matter too much since it's very unlikely somebody will connect a pre-released version of 6.0 to a trusted cluster, but this is how understand why it was done this way before so I did it similarly.

Reply:
It'd also be basically impossible to connect 2 pre-release 6.0 clusters properly since they both would compare "less than" 6.0.0 (which is why I said above this is primarily for development use-case).

Reply:
Ahhh, the semver comparison treats `6.0.0-rc.1` or `-dev` as "pre-release" and that is considered less than `6.0.0`. I assumed it ignored the suffix, my bad.